packages/celotool/src/e2e-tests/slashing_tests.ts

```diff
@@ -0,0 +1,249 @@
+      assert.equal(blockNumber, 1)
+    })
+
+    it('should parse blockNumber from current header', async () => {
+      const contract = await kit._web3Contracts.getElection()
+      const current = await kit.web3.eth.getBlockNumber()
+      const block = await kit.web3.eth.getBlock(current)
+      const rlpEncodedBlock = rlp.encode(headerArray(kit.web3, block))
+      const blockNumber = await contract.methods.getBlockNumberFromHeader(rlpEncodedBlock).call()
+      assert.equal(blockNumber, current)
+    })
+
+    it('should hash test header correctly', async () => {
+      const contract = await kit._web3Contracts.getElection()
+      const header = kit.web3.utils.hexToBytes(
+        '0xf901f9a07285abd5b24742f184ad676e31f6054663b3529bc35ea2fcad8a3e0f642a46f7a01dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347948888f1f195afa192cfee860698584c030f4c9db1a0ecc60e00b3fe5ce9f6e1a10e5469764daf51f1fe93c22ec3f9a7583a80357217a0d35d334d87c0cc0a202e3756bf81fae08b1575f286c7ee7a3f8df4f0f3afc55da056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421b90100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008302000001832fefd8825208845c47775c80a00000000000000000000000000000000000000000000000000000000000000000880000000000000000'
+      )
+      const hash = await contract.methods.hashHeader(header).call()
+      assert.equal(hash, '0xf5a450266c77dce47f7698959d8e7019db860ee19a5322b16a853fdf23607100')
+    })
+
+    it('should hash current header correctly', async () => {
+      const contract = await kit._web3Contracts.getElection()
+      const current = await kit.web3.eth.getBlockNumber()
+      const block = await kit.web3.eth.getBlock(current)
+      const rlpEncodedBlock = rlp.encode(headerArray(kit.web3, block))
+      const blockHash = await contract.methods.hashHeader(rlpEncodedBlock).call()
+      assert.equal(blockHash, block.hash)
+    })
+  })
+
+  let doubleSigningBlock: any
+
+  describe('test slashing for downtime', () => {
+    before(async function(this: any) {
+      this.timeout(0) // Disable test timeout
+      await restartWithDowntime()
+    })
+
+    it('slash for downtime', async function(this: any) {
+      this.timeout(0) // Disable test timeout
+      const slasher = await kit._web3Contracts.getDowntimeSlasher()
+      const blockNumber = await web3.eth.getBlockNumber()
+      await waitUntilBlock(blockNumber + 20)
+      doubleSigningBlock = await web3.eth.getBlock(blockNumber + 15)
```

Comment:
Is this testing double signing or slashing?

edit: took me a second, this is clever. Please describe in a comment what you're doing here